bundles/org.openhab.binding.verisure/src/main/java/org/openhab/binding/verisure/internal/VerisureSession.java

```diff
@@ -0,0 +1,901 @@
+        } catch (TimeoutException e) {
+            logger.warn("TimeoutException: {}", e.getMessage(), e);
+        }
+        return 0;
+    }
+
+    private @Nullable <T> T getJSONVerisureAPI(String url, Class<T> jsonClass) {
+        T result = null;
+        logger.debug("HTTP GET: {}", BASEURL + url);
+        try {
+            ContentResponse response = httpClient.GET(BASEURL + url + "?_=" + System.currentTimeMillis());
+            String pattern = "(?m)^\\s*\\r?\\n|\\r?\\n\\s*(?!.*\\r?\\n)";
+            String replacement = "";
+            String content = response.getContentAsString();
+            logger.debug("HTTP Response ({}) Body:{}", response.getStatus(), content.replaceAll(pattern, replacement));
+
+            if (response.getStatus() == HttpStatus.OK_200) {
+                result = gson.fromJson(content, jsonClass);
+            }
+            return result;
+        } catch (ExecutionException e) {
+            logger.warn("Caught ExecutionException {} for URL string {}", e, url);
+        } catch (InterruptedException e) {
+            logger.warn("Caught InterruptedException {} for URL string {}", e, url);
+        } catch (TimeoutException e) {
+            logger.warn("Caught TimeoutException {} for URL string {}", e, url);
+        }
+        return null;
+    }
+
+    private @Nullable ContentResponse postVerisureAPI(String url, String data, Boolean isJSON) {
+        try {
+            logger.debug("postVerisureAPI URL: {} Data:{}", url, data);
+            Request request = httpClient.newRequest(url).method(HttpMethod.POST);
+            if (isJSON) {
+                request.header("content-type", "application/json");
+            } else {
+                if (csrf != null) {
+                    request.header("X-CSRF-TOKEN", csrf);
+                }
+            }
+            request.header("Accept", "application/json");
+            if (!data.equals("empty")) {
+                request.content(new BytesContentProvider(data.getBytes(StandardCharsets.UTF_8)),
+                        "application/x-www-form-urlencoded; charset=UTF-8");
+            } else {
+                logger.debug("Setting cookie with username {} and vid {}", userName, password);
+                request.cookie(new HttpCookie("username", userName));
+                request.cookie(new HttpCookie("vid", password));
+            }
+            logger.debug("HTTP POST Request {}.", request.toString());
+            return request.send();
+        } catch (ExecutionException e) {
+            logger.warn("Caught ExecutionException {}", e.getMessage(), e);
+        } catch (InterruptedException e) {
+            logger.warn("Caught InterruptedException {}", e.getMessage(), e);
+        } catch (TimeoutException e) {
+            logger.warn("Caught TimeoutException {}", e.getMessage(), e);
+        } catch (RuntimeException e) {
+            logger.warn("Caught RuntimeException {}", e.getMessage(), e);
+        }
```

Comment:
If you aren't going to handle these exceptions in any special way, just group them into a single catch statement. Please do the same in other parts of the code.
```suggestion
        } catch (ExecutionException | InterruptedExceptione | TimeoutException e) {
            logger.warn("Caught Exception {}", e.getMessage(), e);
        }
```

Reply:
Fixed!